tests/integration/targets/aci_l3out/tasks/main.yml

```diff
@@ -0,0 +1,228 @@
+    assert:
+      that:
+        - nm_add_l3out_import_export is changed
+        - nm_add_l3out_import_export.sent.l3extOut.attributes.enforceRtctrl == "export,import"
+        - nm_add_l3out_export is changed
+        - nm_add_l3out_export.sent.l3extOut.attributes.enforceRtctrl == "export"
+        - nm_add_l3out_import_error.msg == "The route_control parameter is invalid: allowed options are export or import,export only"
+
+  - name: Add BGP protocol
+    aci_l3out:
+      <<: *aci_l3out_present
+      l3protocol: bgp
+    register: nm_add_l3out_bgp
+
+  - name: Add BGP protocol again - L3protocol changes work
+    aci_l3out:
+      <<: *aci_l3out_present
+      l3protocol: bgp
+    register: nm_add_l3out_bgp_again
+
+  - name: Add PIM protocol and asn - warning message works for asn
+    aci_l3out:
+      <<: *aci_l3out_present
+      l3protocol: pim
+      asn: 1
+    register: nm_add_l3out_pim
+
+  - name: Add a new L3Out with EIGRP protocol
+    aci_l3out:
+      <<: *aci_l3out_present
+      l3protocol: eigrp
+      asn: 1
+    register: nm_add_l3out_eigrp
+
+  - name: remove asn - failure message works
+    aci_l3out:
+      <<: *aci_l3out_present
+      l3protocol: eigrp
+      asn: null
+    register: add_l3out_without_asn
+    ignore_errors: true
+  
+  - name: remove protocol - static protocol works
+    aci_l3out:
+      <<: *aci_l3out_present
+      l3protocol: static
+    register: nm_remove_l3protocol
+
+  - name: asserts for l3out protocols tasks
+    assert:
+      that:
+        - nm_add_l3out_bgp is changed
+        - nm_add_l3out_bgp_again is changed
+        - nm_add_l3out_pim is changed
+        - nm_add_l3out_eigrp is changed
+        - nm_add_l3out_eigrp.sent.l3extOut.children.1.eigrpExtP.attributes.asn == "1"
+        - add_l3out_without_asn.msg == "Parameter 'asn' is required when l3protocol is 'eigrp'"
+        - nm_remove_l3protocol is changed
+
+  - name: Add OSPF protocol
+    aci_l3out:
+      <<: *aci_l3out_present
+      l3protocol: ospf
+      ospf:
+        area_cost: 1
+        area_ctrl: [ summary, redistribute ]
+        area_id: 0.0.0.1
+        area_type: regular
+        multipod_internal: no
+        description: test for ospf protocol
+    register: nm_add_l3out_ospf
+
+  - name: asserts for l3out OSPF protocol tasks
+    assert:
+      that:
+        - nm_add_l3out_ospf is changed
+        - nm_add_l3out_ospf.sent.l3extOut.children.2.ospfExtP.attributes.areaCost == "1"
+        - nm_add_l3out_ospf.sent.l3extOut.children.2.ospfExtP.attributes.areaCtrl == "summary,redistribute"
+        - nm_add_l3out_ospf.sent.l3extOut.children.2.ospfExtP.attributes.areaId == "0.0.0.1"
+        - nm_add_l3out_ospf.sent.l3extOut.children.2.ospfExtP.attributes.areaType == "regular"
+        - nm_add_l3out_ospf.sent.l3extOut.children.2.ospfExtP.attributes.descr == "test for ospf protocol"
+        - nm_add_l3out_ospf.sent.l3extOut.children.2.ospfExtP.attributes.multipodInternal == "no"
+
+  - name: get all in l3out
```

Comment:
Can you add multiple l3outs and do a query that retrieves more than 1 result.

Reply:
yes, I can and will do.